handel/src/evaluator.rs

```diff
@@ -256,13 +247,18 @@ where
         votes >= self.threshold
     }
 
-    fn level_contains_id(&self, level: usize, id: usize) -> bool {
-        if level == self.partitioner.levels() {
-            return true;
+    fn level_contains_origin(&self, msg: &LevelUpdate<TProtocol::Contribution>) -> bool {
+        if msg.level as usize == self.partitioner.levels() {
+            let weight = self
+                .weights
+                .signers_identity(&msg.aggregate.contributors())
+                .len();
+            // Only available to full aggregations
+            return weight == self.partitioner.size();
         }
-        let range = self.partitioner.range(level);
+        let range = self.partitioner.range(msg.level as usize);
```

Comment:
I believe that is going to be the case in a couple more places. Would you say this would be more of a cosmetic change or do you see a more functional benefit?

I would do it later if it is considered beneficial.

Reply:
Cosmetic. Our levels don't get large enough to get in the range of the platform differences.